### ISSUE TITLE
have_many :as tests

### ISSUE THREAD

**Issue by amnesia7** [thoughtbot/shoulda-matchers#735]
I was wondering if `:as` should be added to `have_many`.

I have 

```
has_many :favourites, as: :favourable, dependent: :destroy
```

and am testing it using:

```
it { should have_many(:favourites).dependent(:destroy) }
```

Since `:as` isn't currently available in shoulda-matchers I've also added the following test to my spec file but I'm not sure if I should really be testing anything more as part of `:as`

```
it { should respond_to(:favourable) }
```

Thanks

**Comment by maurogeorge** [comment]
We are talking here about polymorphic associations right?

I think the `it { should have_many(:favourites).dependent(:destroy) }` is enough. Since if you change your associations to `has_many :favourites, dependent: :destroy` you will receive a error like:

```
     Failure/Error: it { is_expected.to have_many(:favourites).dependent(:destroy) }
       Expected Model to have a has_many association called favourites (Favourite does not have a model_id foreign key.)
```
Maybe we can add a note on the docs, like we have on the [belong_to](http://matchers.shoulda.io/docs/v2.8.0/Shoulda/Matchers/ActiveRecord.html#belong_to-instance_method).

What you think?

**Comment by amnesia7** [comment]
Just tested without `:as` and failed as you say so a note on the docs sounds fine to me.

**Comment by mcmire** [comment]
Yeah, I think a note in the docs makes sense here.

**Comment by mcmire** [comment]
Fixed in [670f154](https://github.com/thoughtbot/shoulda-matchers/commit/670f154afcc6363ebcb415d15f589140ebb438ea).
